Magento_Cms/templates/custom-country-form.phtml

```diff
@@ -0,0 +1,25 @@
+<?php
+/** @var \Magento\Framework\View\Element\Template $block */
+/** @var \Magento\Framework\Escaper $escaper */
+?>
+
+<h3>Search Country</h3>
+<p id="error-message"></p>
+<form id="search-country-form" data-mage-init='{"validation": {}}'>
+    <label for="country-id">Country ID</label>
+    <input type="text" name="country-id" id="country-id" class="country-id" data-validate="{'validate-country-id':true}" placeholder="Country ID..."/>
+    <button type="submit">Search</button>
+</form>
+
+<div>
+    <p id="country-name"></p>
```

Comment:
you use a lot of `id` attributes. what will happen if your form is duplicated twice or more? do you remember issues with the discount code validation form? 
I recommend using classes with `js-` prefix, like
```html
<p class="js-country-name country-name"></p>
```
js-country-name is for JS functionality
country-name is for CSS styling

the same for other elements